tracetools/src/tracetools.c

```diff
@@ -489,6 +491,17 @@ DEFINE_TRACEPOINT(
   TRACEPOINT_ARGS(
     buffer))
 
+void __attribute__((constructor)) tracetools_init()
+{
+  if(getenv("TRACETOOLS_DISABLE")) {
+    fprintf(stderr, "ROS 2 tracing disabled\n");
+    return;
+  }
+  if(!dlopen("libtracetools_provider.so", RTLD_NOW | RTLD_GLOBAL)) {
+    fprintf(stderr, "Failed to load tracepoint provider for ROS 2 tracing, disabling\n");
```

Comment:
I'm assuming you probably won't keep the prints here, but I'd like to have something like it and include this mechanism in the [status tool (`ros2 run tracetools status`)](https://github.com/ros2/ros2_tracing#building):

```
$ TRACETOOLS_DISABLE=y ros2 run tracetools status 
ROS 2 tracing disabled
ROS 2 tracing disabled
Tracing enabled
```

I know it's not fully representative (e.g., if you set `TRACETOOLS_DISABLE=y` for your app but not `ros2 run tracetools status`), but I think it would still be useful if it's used in the same environment as an application. What do you think?

Reply:
Hmm I think it makes sense for the `ros2 run tracetools status` to report on it. I also think it makes sense to log something for the user. `stderr` is perhaps a bit raw. I could try and use `rcutils` logging capabilities, and also expose some flag for `ros2 run tracetools status` to check and provide a better reporting. Let me give this a shot.

Reply:
Hmm, no, I don't like the `rcutils` dependency. I'll find another way.

Reply:
Sticking to `fprintf` with a verbosity flag. See 814bdd421e14cb4a6a469719edbe1e39f96fab99.

Reply:
`TRACETOOLS_VERBOSE` looks good.

> Hmm, no, I don't like the rcutils dependency. I'll find another way.

Agreed.